packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/ExecuteRulesTests.java

```diff
@@ -50,47 +63,265 @@
+
+        STEP("Check if Audio and Lockable aspects were added to child folder's file");
+        fileNode = restClient.authenticateUser(user).withCoreAPI().usingNode(childFolderFile).getNode();
+        restClient.assertStatusCodeIs(HttpStatus.OK);
+        assertThat(fileNode)
+            .containsAspects(AUDIO_ASPECT, LOCKABLE_ASPECT);
+        // LOCKABLE_ASPECT shouldn't be present here, see ACS-3683
+    }
+
+    /**
+     * Try to execute disabled rule and check if nothing changed.
+     */
+    @Test(groups = { TestGroup.REST_API, TestGroup.RULES, TestGroup.ACTIONS })
+    public void executeRules_disabledRule()
+    {
+        STEP("Disable folder rules");
+        childFolderRule.setIsEnabled(false);
+        restClient.authenticateUser(user).withCoreAPI().usingNode(childFolder).usingDefaultRuleSet().updateRule(childFolderRule.getId(), childFolderRule);
+
         STEP("Check if file aspects don't contain Audio one");
-        RestNodeModel node = restClient.authenticateUser(user).withCoreAPI().usingNode(file).getNode();
+        RestNodeModel fileNode = restClient.authenticateUser(user).withCoreAPI().usingNode(childFolderFile).getNode();
         restClient.assertStatusCodeIs(HttpStatus.OK);
-        node.assertThat().field("aspectNames").notContains(AUDIO_ASPECT);
+        assertThat(fileNode).notContainsAspects(AUDIO_ASPECT);
 
         STEP("Execute rule");
-        restClient.authenticateUser(user).withCoreAPI().usingNode(ruleFolder).executeRules(createRuleExecutionRequest());
+        RestRuleExecutionModel executionResult = restClient.authenticateUser(user).withCoreAPI().usingNode(childFolder).executeRules(createRuleExecutionRequest());
         restClient.assertStatusCodeIs(HttpStatus.CREATED);
+        executionResult.assertThat().field("isEachSubFolderIncluded").is(false);
+        executionResult.assertThat().field("isEachInheritedRuleExecuted").is(false);
 
-        STEP("Check if file contains Audio aspect");
-        Utility.sleep(500, 10000, () -> {
-            RestNodeModel fileNode = restClient.authenticateUser(user).withCoreAPI().usingNode(file).getNode();
-            restClient.assertStatusCodeIs(HttpStatus.OK);
-            fileNode.assertThat().field("aspectNames").contains(AUDIO_ASPECT);
-        });
+        STEP("Check if Audio aspect is still missing");
+        fileNode = restClient.authenticateUser(user).withCoreAPI().usingNode(childFolderFile).getNode();
+        restClient.assertStatusCodeIs(HttpStatus.OK);
+        assertThat(fileNode).notContainsAspects(AUDIO_ASPECT);
     }
 
-    // TODO add more E2Es. For more see: ACS-3620
+    /**
+     * Try to execute inherited parent folder's rule which is not inheritable.
+     */
+    @Test(groups = { TestGroup.REST_API, TestGroup.RULES, TestGroup.ACTIONS })
+    public void executeRules_notInheritableRule()
+    {
+        STEP("Set parent rule as not inheritable");
+        parentFolderRule.setIsInheritable(false);
```

Comment:
Please create a new rule if you're going to change it. This will affect other tests in the suite.

Reply:
This shouldn't affect other tests as rules are recreated before every test method but I fixed it have it more clear.